clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -792,6 +792,9 @@ public void sendOffsetsToTransaction(Map<TopicPartition, OffsetAndMetadata> offs
      * <p>
      * Further, if any of the {@link #send(ProducerRecord)} calls which were part of the transaction hit irrecoverable
      * errors, this method will throw the last received exception immediately and the transaction will not be committed.
+     * It should be noted that if <code>flush()</code> is called explicitly beforehand, this method will NOT throw any
+     * exception related to the {@link #send(ProducerRecord)} calls. Since <code>flush()</code> clears the last received
+     * exception and returns the transaction from the error state.
```

Comment:
> returns the transaction from the error state.

The word "returns" is confusing to me -- but I am not a native speaker. Should it be "and transits the transaction out of any error state"

Reply:
>Should it be "and transits the transaction out of any error state"
I agree with you.